muxink/src/codec/bytesrepr.rs

```diff
@@ -0,0 +1,172 @@
+//! Bytesrepr encoding/decoding
+use std::{fmt::Debug, marker::PhantomData};
+
+use bytes::{Buf, Bytes, BytesMut};
+use casper_types::bytesrepr::{self, FromBytes, ToBytes};
+use thiserror::Error;
+
+use super::{DecodeResult, FrameDecoder, Transcoder};
+use crate::codec::DecodeResult::Failed;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("bytesrepr error")]
+    BytesreprError(bytesrepr::Error),
+}
+
+/// A bytesrepr encoder.
+#[derive(Default)]
+pub struct BytesreprEncoder<T> {
+    /// Item type processed by this encoder.
+    ///
+    /// We restrict encoders to a single message type to make decoding on the other end easier.
+    item_type: PhantomData<T>,
+}
+
+impl<T> BytesreprEncoder<T> {
+    /// Creates a new bytesrepr encoder.
+    pub fn new() -> Self {
+        BytesreprEncoder {
+            item_type: PhantomData,
+        }
+    }
+}
+
+impl<T> Transcoder<T> for BytesreprEncoder<T>
+where
+    T: ToBytes,
+{
+    type Error = Error;
+
+    type Output = Bytes;
+
+    fn transcode(&mut self, input: T) -> Result<Self::Output, Self::Error> {
+        Ok(input.to_bytes().map_err(Error::BytesreprError)?.into())
```

Comment:
I think this should use `into_bytes` for efficiency.

Reply:
Updated in 44b1fa3b0b4c2f02fb801923920fae3d4c805ad7